street_network/src/lib.rs

```diff
@@ -216,6 +241,44 @@ impl StreetNetwork {
             Ok(results.trimmed_center_pts.remove(&road_id).unwrap().0)
         }
     }
+
+    pub(crate) fn copy_for_debugging<I: Into<String>>(&self, label: I) -> DebugStreets {
+        DebugStreets {
+            label: label.into(),
+            streets: StreetNetwork {
+                roads: self.roads.clone(),
+                intersections: self.intersections.clone(),
+                boundary_polygon: self.boundary_polygon.clone(),
+                gps_bounds: self.gps_bounds.clone(),
+                config: self.config.clone(),
+                debug_steps: RefCell::new(Vec::new()),
+                debug_points: RefCell::new(Vec::new()),
+                debug_polylines: RefCell::new(Vec::new()),
+            },
+            points: self.debug_points.borrow_mut().drain(..).collect(),
+            polylines: self.debug_polylines.borrow_mut().drain(..).collect(),
+        }
+    }
+
+    pub(crate) fn debug_intersection<I: Into<String>>(&self, i: osm::NodeID, label: I) {
+        if self.debug_steps.borrow().is_empty() {
```

Comment:
Short-circuit and don't write anything if debugging is disabled. We push a copy of the original `StreetNetwork` on at the beginning of `apply_transformations_stepwise_debugging`, thus indicating we're in a debug mode